mettle/worker/worker.go

```diff
@@ -915,6 +933,22 @@ func (w *worker) update(stage pb.ExecutionStage_Value, response *pb.ExecuteRespo
 	return common.PublishWithOrderingKey(ctx, w.responses, body, w.actionDigest.Hash, w.name)
 }
 
+// periodicallyPushMetrics will push this worker's metrics to the gateway every 5 minutes.
+func (w *worker) periodicallyPushMetrics() {
+	if w.promGatewayURL != "" {
+		for {
+			for metric, metricName := range metrics {
```

Comment:
This doesn't look quite right, I don't think you should have to create a new Collector for every metric each time around the loop? I used to have some code in plz that did this, it's from a while back but I don't recall it doing it quite like this 🤔